spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/actions/BaseExpireSnapshotsSparkAction.java

```diff
@@ -170,13 +177,15 @@ public BaseExpireSnapshotsSparkAction deleteWith(Consumer<String> newDeleteFunc)
         expireSnapshots = expireSnapshots.retainLast(retainLastValue);
       }
 
+      List<Snapshot> expired = expireSnapshots.apply();
+      Set<Long> expiredSnapshotIds = expired.stream().map(Snapshot::snapshotId).collect(Collectors.toSet());
```

Comment:
I think this needs to be done by diffing the snapshot ID sets between metadata versions.

Reply:
Done